sycl/source/detail/scheduler/commands.cpp

```diff
@@ -930,9 +930,16 @@ cl_int ExecCGCommand::enqueueImp() {
       case kernel_param_kind_t::kind_accessor: {
         Requirement *Req = (Requirement *)(Arg.MPtr);
         AllocaCommandBase *AllocaCmd = getAllocaForReq(Req);
+#if USE_PI_CUDA
+        pi_mem MemArg = (pi_mem)AllocaCmd->getMemAllocation();
+        Plugin.call<PiApiKind::piextKernelSetArgMemObj>(Kernel, Arg.MIndex, &MemArg);
+#else
         cl_mem MemArg = (cl_mem)AllocaCmd->getMemAllocation();
         Plugin.call<PiApiKind::piKernelSetArg>(Kernel, Arg.MIndex,
                                                sizeof(cl_mem), &MemArg);
+        Plugin.call<PiApiKind::piKernelSetArg>(Kernel, Arg.MIndex,
```

Comment:
@bader @Alexander-Johnston Was this extra API added by mistake?
Is there a reason for adding it twice?
FYI @rbegam .